core-initrd/latest/factory/usr/lib/systemd/system/sysroot.mount

```diff
@@ -1,21 +1,16 @@
+# This file is overridden by the one created by snap-bootstrap in
```

Comment:
What really depends on it? Cannot we just change the dependencies of those? I do not see anything from systemd that depends on it directly. Things should only depend on initrd-root-fs.target. So we could just make snap-initramfs-mounts.service to run before initrd-root-fs.target

Reply:
snap-bootstrap is not mounting `sysroot.mount`, it just overrides the unit is `/usr/lib/systemd/system/`, but relies on the currently defined dependencies to get the unit started.

But, we could try to run `systemd-mount` to force it to be mounted when snap-bootstrap is running and then maybe get rid of this file.

There are some services that depend on this unit:

- populate-writable.service
- sysroot-writable.mount
- systemd-volatile-root.service (we don't use this one)
- Also initrd-root-fs.target wants it.

These units have a `Before=initrd-root-fs.target`, but maybe it could be replaced by `Before=initrd-parse-etc.service`. Or we can keep them as they are if using systemd-mount works.

Reply:
snap-initramfs-mounts.service should run "Before" initrd-root-fs.target. And then it will daemon-reload and inject sysroot.mount as Before initrd-root-fs.target.